lib/ruby_lsp/ruby_document.rb

```diff
@@ -257,6 +257,37 @@ def locate_node(position, node_types: [])
       )
     end
 
+    sig { returns(String) }
+    def test_library
+      return "none" unless test_file?
+
+      class_entries = @global_state.index.entries_for(@uri.to_s, RubyIndexer::Entry::Class)
+      return "unknown" unless class_entries
+
+      # TODO: consider performance hit
+      # A better approach might be check the classes entries one at a time.
+      ancestors = class_entries
+        .reject { T.cast(_1, RubyIndexer::Entry::Class).parent_class == "::Object" }
+        .map { @global_state.index.linearized_ancestors_of(_1.name) }.flatten
+
+      # ActiveSupport::TestCase is a subclass of Minitest::Test so we must check for it first
+      if ancestors.include?("ActiveSupport::TestCase")
+        "rails"
+      elsif ancestors.include?("Minitest::Test")
+        "minitest"
+      elsif ancestors.include?("Test::Unit::TestCase")
+        "test-unit"
+      else
+        "unknown"
```

Comment:
We should become a bit more specific now that we're checking for ancestors. For example, as [Alex mentioned here](https://github.com/Shopify/ruby-lsp/issues/1334#issuecomment-2173793731), the test framework isn't really "rails" and is not tied to `ActiveSupport::TestCase`.

We can check for `ActiveSupport::Testing::Declarative` and cover a broader range of cases.

Same thing for Minitest and Test Unit. We can now do a better job at detecting the spec style syntax.